Unable to obtain publisher GID and sequence number errors

Hi,

Thanks for all your work on the Zenoh RMW! We are very excited to try it out. 

I'm getting the following errors when using data from a rosbag through `ros2 bag play`. These errors show up in any node that subscribes to topics published over Zenoh. It seems like these errors go away when subscribing to "live" topics rather than those played back through `ros2 bag`. Do you have any idea what the root cause could be? 

```
[ERROR] [1710953784.261021973] [rmw_zenoh_cpp]: Unable to obtain sequence number from the attachment.
[ERROR] [1710953784.261040865] [rmw_zenoh_cpp]: Unable to obtain publisher GID from the attachment.
```

--- a/rmw_zenoh_cpp/src/rmw_zenoh.cpp
+++ b/rmw_zenoh_cpp/src/rmw_zenoh.cpp
@@ -1044,6 +1044,20 @@ rmw_publish_serialized_message(
     return RMW_RET_ERROR;
   }
 
+  uint64_t sequence_number = publisher_data->get_next_sequence_number();
+
+  z_owned_bytes_map_t map = 
+    create_map_and_set_sequence_num(sequence_number, publisher_data->pub_gid);
+
+  if (!z_check(map)) {
+    // create_map_and_set_sequence_num already set the error
+    return RMW_RET_ERROR;
+  }
+  auto free_attachment_map = rcpputils::make_scope_exit(
+    [&map]() {
+      z_bytes_map_drop(z_move(map));
+    });
+
   const size_t data_length = ser.getSerializedDataLength();
 
   // The encoding is simply forwarded and is useful when key expressions in the
@@ -1051,6 +1065,7 @@ rmw_publish_serialized_message(
   // will be encoded with CDR so it does not really matter.
   z_publisher_put_options_t options = z_publisher_put_options_default();
   options.encoding = z_encoding(Z_ENCODING_PREFIX_EMPTY, NULL);
+  options.attachment = z_bytes_map_as_attachment(&map);
   // Returns 0 if success.
   int8_t ret = z_publisher_put(
     z_loan(publisher_data->pub),

--- a/rmw_zenoh_cpp/src/rmw_zenoh.cpp
+++ b/rmw_zenoh_cpp/src/rmw_zenoh.cpp
@@ -1046,7 +1046,7 @@ rmw_publish_serialized_message(
 
   uint64_t sequence_number = publisher_data->get_next_sequence_number();
 
-  z_owned_bytes_map_t map = 
+  z_owned_bytes_map_t map =
     create_map_and_set_sequence_num(sequence_number, publisher_data->pub_gid);
 
   if (!z_check(map)) {

